compiler/damlc/daml-compiler/src/DA/Daml/Compiler/Upgrade.hs

```diff
@@ -112,8 +112,7 @@ generateTemplateInstanceModule env externPkgId
               map (showSDocForUser fakeDynFlags alwaysQualify . ppr) instances)
     | otherwise = Nothing
   where
-    instances = templInstances
-    templInstances = templateInstances env externPkgId
+    instances = templateInstances env externPkgId ++ choiceInstances env externPkgId
```

Comment:
btw, in case you’re still looking for what to work on next, we also need `TemplateKey` instances.

Reply:
I'll discuss with Robin who is best placed to work on these.